File: JavaApp/src/main/java/com/charlezz/javaapp/feature/remote/PostService.java
package com.charlezz.javaapp.feature.remote;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface PostService {
    @GET("posts")
    Call<List<Post>> getPosts(@Query("_page") int page);

    @GET("posts?_page=1")
    Call<List<Post>> getTopPosts();
}

File: app/src/main/java/com/charlezz/android/network/PostService.kt
package com.charlezz.android.network

import com.charlezz.android.db.entity.Post
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface PostService {

    @GET("?rest_route=/wp/v2/posts")
    suspend fun getPosts(): List<Post>

    @GET("?rest_route=/wp/v2/posts")
    fun getPostsByPage(@Query("page") page: Int): List<Post>

    @GET("?rest_route=/wp/v2/posts")
    suspend fun getPostsBefore(@Query("before") before: String, @Query("per_page") perPage: Int): List<Post>

    @GET("?rest_route=/wp/v2/posts")
    suspend fun getPostsAfter(@Query("after") after: String, @Query("per_page") perPage: Int): List<Post>

    @GET("?rest_route=/wp/v2/posts")
    fun getPosts(
            @Query("after") after: String?,
            @Query("before") before: String?,
            @Query("per_page") perPage: Int = 10
    ): Call<List<Post>>

}